src/components/Canvas.tsx

```diff
@@ -17,25 +17,28 @@ interface CanvasProps {
   data: string;
   selectedBits: BitSelection;
   isGridView: boolean;
+  isDisplay: boolean; // New prop for play/pause functionality
 }
 
-const Canvas: React.FC<CanvasProps> = ({ data, selectedBits, isGridView }) => {
-  const { theme } = useTheme(); // Get the current theme
-
-  const channels = useMemo(() => [true, true, true, true], []); // Number of channels
-
-  const [isPaused, setIsPaused] = useState(Array(channels.length).fill(false)); // Paused state for each channel
-  const chartRef = useRef<SmoothieChart[]>([]); // Reference to the chart
-  const seriesRef = useRef<(TimeSeries | null)[]>([]); // Reference to the timeseries
-  const [isChartInitialized, setIsChartInitialized] = useState(false); // Chart initialization state
-  const batchSize = 10; // Batch size for processing data
-  const batchBuffer = useMemo<Array<{ time: number; values: number[] }>>( // Buffer for batch processing
+const Canvas: React.FC<CanvasProps> = ({
+  data,
+  selectedBits,
+  isGridView,
+  isDisplay,
+}) => {
+  const { theme } = useTheme();
+  const channels = useMemo(() => [true, true, true, true], []);
+  const chartRef = useRef<SmoothieChart[]>([]);
+  const seriesRef = useRef<(TimeSeries | null)[]>([]);
+  const [isChartInitialized, setIsChartInitialized] = useState(false);
+  const [isPaused, setIsPaused] = useState(Array(channels.length).fill(false)); // Paused state for each channe
```

Comment:
Just fill this array with isAllPaused. When isAllPaused will be true, the array will get filled by true values and vice versa, after this we can handle individual pause and resume too.